### PR TITLE
fzf: option for tmux usage in shell integration

### PR DESCRIPTION
**Pull request by afreakk**
### Description

Added options for tmux which will be read by fzf shell integrations.
The env variables are not read by the fzf binary, they only change behaviour of fzf shell integrations.
(Like FZF_CTRL_*_* and FZF_ALT_*_* env variables.)

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
(there arent any test cases for fzf)

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/fzf.nix [hunk 1]
```diff
@@ -88,6 +88,25 @@ in {
...
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "-d 40%" ];
+      description = ''
+        If useTmuxInShellIntegration is set to true,
+        shell integration will use these options for fzf-tmux.
+        See fzf-tmux --help for available options.
+      '';
+    };
+
```
Comment (berbiche) [inline modules/programs/fzf.nix:109]: Could these options be nested under a `tmux` setting?

```suggestion
tmux = {
    enableShellIntegration = mkEnableOption ''
        setting FZF_TMUX to 1 which causes shell integration to use fzf-tmux
      '';
    };

    shellIntegrationOptions = mkOption {
      type = types.listOf types.str;
      default = [ ];
      example = [ "-d 40%" ];
      description = ''
        If <options>programs.fzf.tmux.enableShellIntegration<options> is set to true,
        shell integration will use these options for fzf-tmux.
        See fzf-tmux --help for available options.
      '';
    };
}
```

File: modules/programs/fzf.nix [hunk 2]
```diff
@@ -125,6 +144,8 @@ in {
         FZF_CTRL_T_OPTS = cfg.fileWidgetOptions;
         FZF_DEFAULT_COMMAND = cfg.defaultCommand;
         FZF_DEFAULT_OPTS = cfg.defaultOptions;
+        FZF_TMUX = if cfg.useTmuxInShellIntegration then "1" else null;
+        FZF_TMUX_OPTS = cfg.tmuxOptionsInShellIntegration;
```
Comment (berbiche) [inline modules/programs/fzf.nix:148]:
```suggestion
        FZF_TMUX = if cfg.tmux.enableShellIntegration then "1" else null;
        FZF_TMUX_OPTS = cfg.tmux.shellIntegrationOptions;
```

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: modules/programs/fzf.nix [hunk 3]
```diff
@@ -88,6 +88,23 @@ in {
       '';
     };
 
+    tmux = {
+      enableShellIntegration = mkEnableOption ''
+        sets FZF_TMUX=1 which causes shell integration to use fzf-tmux
```
Comment (berbiche) [inline modules/programs/fzf.nix:93]: This expands to `Whether to enable sets FZF_TMUX=1`.
I recommend the initial `setting <literal>FZF_TMUX=1</literal>`.

File: modules/programs/fzf.nix [hunk 4]
```diff
@@ -88,6 +88,23 @@ in {
...
+      '';
+
+      shellIntegrationOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "-d 40%" ];
+        description = ''
+          If <options>programs.fzf.tmux.enableShellIntegration<options> is set to true,
+          shell integration will use these options for fzf-tmux.
+          See fzf-tmux --help for available options.
```
Comment (berbiche) [inline modules/programs/fzf.nix:103]:
```suggestion
          See <command>fzf-tmux --help</command> for available options.
```

**Review by berbiche — CHANGES_REQUESTED** [review]
This needs some small documentation fixes and it's good 👍

File: modules/programs/fzf.nix [hunk 5]
```diff
@@ -88,6 +88,23 @@ in {
...
 
+    tmux = {
+      enableShellIntegration = mkEnableOption ''
+        setting <literal>FZF_TMUX=1</literal> which causes shell integration to use fzf-tmux
+      '';
+
+      shellIntegrationOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "-d 40%" ];
```
Comment (berbiche) [inline modules/programs/fzf.nix:99]:
```suggestion
        example = literalExample ''[ "-d 40%" ]'';
```
One last thing to fix.

File: modules/programs/fzf.nix [hunk 6]
```diff
@@ -88,6 +88,23 @@ in {
...
+      enableShellIntegration = mkEnableOption ''
+        setting <literal>FZF_TMUX=1</literal> which causes shell integration to use fzf-tmux
+      '';
+
+      shellIntegrationOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = literalExample ''[ "-d 40%" ]'';
+        description = ''
+          If <options>programs.fzf.tmux.enableShellIntegration<options> is set to true,
```
Comment (sumnerevans) [inline modules/programs/fzf.nix:101]: The build is failing due to the end tag being incorrect. Also, I think it should be `option` not `options`.

```suggestion
          If <option>programs.fzf.tmux.enableShellIntegration</option> is set to true,
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Pull request merged**